ssc2museca.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import datetime
+import os
+import sys
+
+from typing import Dict, Any, List, Tuple, Optional
+from xml.dom import minidom  # type: ignore
+
+from chartv2 import Chartv2, XMLv2
+from audio import TwoDX, ADPCM
+
+def main() -> int:
+    parser = argparse.ArgumentParser(description="A utility to convert 16-lane StepMania charts (.ssc format) to Museca format.")
+    parser.add_argument(
+        "file",
+        metavar="FILE",
+        help=".mu file to convert to Museca format.",
+        type=str,
+    )
+    parser.add_argument(
+        "id",
+        metavar="ID",
+        help="ID to assign this song.",
+        type=int,
+    )
+    parser.add_argument(
+        "-d",
+        "--directory",
+        help="Directory to place files in. Defaults to current directory.",
+        default='.',
+    )
+    parser.add_argument(
+        "-x",
+        "--update-xml",
+        help="Location of music database XML file to update. If not specified, "
+             "a new one will be created containing just the data for this file.",
+        default=None,
+    )
+
+    args = parser.parse_args()
+    root = args.directory
+    if root[-1] != '/':
+        root = root + '/'
+    root = os.path.realpath(root)
+    os.makedirs(root, exist_ok=True)
+
+    fp = open(args.file, 'rb')
+    data = fp.read()
+    fp.close()
+
+    print('Parsing chart data...')
+
+    # First, parse out the chart and get the XML writer ready.
+    chart = Chartv2(data)
+    xml = XMLv2(chart, args.id)
+
+    print('Outputting XML...')
+
+    if args.update_xml is not None:
+        # First, update the metadata file with the info from this chart.
+        fp = open(args.update_xml, "rb")
+        data = fp.read()
+        fp.close()
+        fp = open(args.update_xml, "wb")
+        fp.write(xml.update_metadata(data))
+        fp.close()
+    else:
+        # First, write out a metadata file, that can be copied into music-info.xml
+        fp = open(os.path.join(root, 'music-info.xml'), 'wb')
+        fp.write(xml.get_metadata())
+        fp.close()
+
+    # Now, write out the chart data - nov/green
+    fp = open(os.path.join(root, '01_{num:04d}_nov.xml'.format(num=args.id)), 'wb')
+    fp.write(xml.get_notes('easy'))
+    fp.close()
+
+    # Now, write out the chart data - adv/yellow
+    fp = open(os.path.join(root, '01_{num:04d}_adv.xml'.format(num=args.id)), 'wb')
+    fp.write(xml.get_notes('medium'))
+    fp.close()
+
+    # Now, write out the chart data - exh/red
+    fp = open(os.path.join(root, '01_{num:04d}_exh.xml'.format(num=args.id)), 'wb')
+    fp.write(xml.get_notes('hard'))
+    fp.close()
+
+    # Should we even attempt inf?
```

Comment:
No on this, the game doesn't support it.

Reply:
Alright, crossing that one off the "experiments pending" list. Lack of any menu textures for it made me lean heavily toward "this won't work".